Add a secondary control plane LB

nrb opened the pull request:
Current issues:

* Fuzzy conversion tests aren't passing
* AWS Machine controller needs to account for both LBs
* Webhook validation tests will fail because both new and old LBSpecs are `nil`

Fedosin committed:
Allow to configure garbage collector using clusterawsadm

AndiDog committed:
Tag S3 bucket as owned by cluster

richardcase committed:
admin: changes to reviewers and updating owner aliases

Signed-off-by: Richard Case <richard.case@outlook.com>

vincepri committed:
:seedling: Allow ESP protocol to be set in IngressRules

Signed-off-by: Vince Prignano <vincepri@redhat.com>

vincepri committed:
:sparkles: Support ignition v3 and presigned URLs

Signed-off-by: Vince Prignano <vincepri@redhat.com>

nrb committed:
Support a secondary control plane load balancer

* A secondary control plane load balancer can be added.
* It will be a network load balancer.
* Its scheme must not match that of the ControlPlaneLoadBalancer.

Previously, NLBs could not have security groups attached. This has now
changed, and an NLB can have a security group attached at creation. If a
security group is _not_ present at creation, then the NLB can never have
security groups added.

vincepri committed:
:bug: On deletion, should ignore if the secret cannot be found

Signed-off-by: Vince Prignano <vincepri@redhat.com>

richardcase committed:
chore: move dthorsen and pydctw to emeritus

Signed-off-by: Richard Case <richard.case@outlook.com>

dependabot[bot] committed:
build(deps): bump github.com/onsi/ginkgo/v2 from 2.12.1 to 2.13.0

Bumps [github.com/onsi/ginkgo/v2](https://github.com/onsi/ginkgo) from 2.12.1 to 2.13.0.
- [Release notes](https://github.com/onsi/ginkgo/releases)
- [Changelog](https://github.com/onsi/ginkgo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/ginkgo/compare/v2.12.1...v2.13.0)

---
updated-dependencies:
- dependency-name: github.com/onsi/ginkgo/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

JoelSpeed commented on api/v1beta2/awscluster_webhook.go:
```diff
@@ -243,19 +251,37 @@ func (r *AWSCluster) validateNetwork() field.ErrorList {
-	if r.Spec.ControlPlaneLoadBalancer == nil {
-		return allErrs
+	// If the secondary is defined, check that the name is not empty and different from the primary.
+	if r.Spec.SecondaryControlPlaneLoadBalancer != nil {
+		if r.Spec.SecondaryControlPlaneLoadBalancer.Name == nil || *r.Spec.SecondaryControlPlaneLoadBalancer.Name == "" {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "secondaryControlPlaneLoadBalancer", "name"), r.Spec.SecondaryControlPlaneLoadBalancer.Name, "secondary controlPlaneLoadBalancer.name cannot be empty"))
+		}
```
I'm guessing CAPI isn't using CEL validations for their API yet. This is an easy 1 liner in CEL on the secondary field
```
// +kubebuilder:validation:XValidation:rule="has(self.name) && self.name != ''",message="name is required for secondaryControlPlaneLoadBalancer"
```

vincepri committed:
:sparkles: Support additional TCP listeners on NLBs

Signed-off-by: Vince Prignano <vincepri@redhat.com>

vincepri committed:
:bug: Fix "error: externally-managed-environment" CI

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4564 from vincepri/fixup-python-ci

:bug: Fix "error: externally-managed-environment" CI

vincepri committed:
Remove double install

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4565 from vincepri/fixup-e2e

e2e: Remove double install

vincepri committed:
ELBv2 target groups should inherit tags

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4518 from AndiDog/tag-s3-bucket

🌱 Tag S3 bucket as owned by cluster

vincepri committed:
Try to delete as many resources as possible, then requeue

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4553 from vincepri/ignition-presigned

:sparkles: Support ignition v3 and presigned URLs

k8s-ci-robot committed:
Merge pull request #4559 from kubernetes-sigs/dependabot/go_modules/github.com/onsi/ginkgo/v2-2.13.0

build(deps): bump github.com/onsi/ginkgo/v2 from 2.12.1 to 2.13.0

k8s-ci-robot committed:
Merge pull request #4566 from vincepri/elbv2tags

:bug: ELBv2 target groups should inherit tags

dependabot[bot] committed:
build(deps): bump github.com/google/go-cmp from 0.5.9 to 0.6.0

Bumps [github.com/google/go-cmp](https://github.com/google/go-cmp) from 0.5.9 to 0.6.0.
- [Release notes](https://github.com/google/go-cmp/releases)
- [Commits](https://github.com/google/go-cmp/compare/v0.5.9...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-cmp
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4570 from kubernetes-sigs/dependabot/go_modules/github.com/google/go-cmp-0.6.0

build(deps): bump github.com/google/go-cmp from 0.5.9 to 0.6.0

vincepri committed:
When tagging security groups, ensure cloud provide tag is only on lb

Signed-off-by: Vince Prignano <vincepri@redhat.com>

nrb commented on api/v1beta1/awscluster_conversion.go:
```diff
@@ -134,6 +150,10 @@ func restoreControlPlaneLoadBalancer(restored, dst *infrav2.AWSLoadBalancerSpec)
+	dst.AdditionalSecurityGroups = restored.AdditionalSecurityGroups
+	dst.Scheme = restored.Scheme
+	dst.CrossZoneLoadBalancing = restored.CrossZoneLoadBalancing
+	dst.Subnets = restored.Subnets
```
Adding these fields caused conversion tests to pass, but it seems incorrect if they were missing for the control plane before.

nrb commented on api/v1beta2/network_types.go:
```diff
@@ -37,6 +37,9 @@ type NetworkStatus struct {
 	APIServerELB LoadBalancer `json:"apiServerElb,omitempty"`
 
+	// SecondaryAPIServerELB is the secondary Kubernetes api server load balancer.
+	SecondaryAPIServerELB LoadBalancer `json:"secondaryAPIServerELB,omitempty"`
```
Does this need to be present? I assume if we create a resource, we want to track it.

vincepri replied:
Yes, makes sense to me to reflect in status

dependabot[bot] committed:
build(deps): bump golang.org/x/net from 0.14.0 to 0.17.0

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.14.0 to 0.17.0.
- [Commits](https://github.com/golang/net/compare/v0.14.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4572 from kubernetes-sigs/dependabot/go_modules/golang.org/x/net-0.17.0

build(deps): bump golang.org/x/net from 0.14.0 to 0.17.0

dependabot[bot] committed:
build(deps): bump golang.org/x/net from 0.15.0 to 0.17.0 in /hack/tools

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.15.0 to 0.17.0.
- [Commits](https://github.com/golang/net/compare/v0.15.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4573 from kubernetes-sigs/dependabot/go_modules/hack/tools/golang.org/x/net-0.17.0

build(deps): bump golang.org/x/net from 0.15.0 to 0.17.0 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4557 from vincepri/fix-secret-not-found

:bug: On deletion, should ignore if the secret cannot be found

richardcase committed:
wip: eks e2e changes

Signed-off-by: Richard Case <richard.case@outlook.com>

richardcase committed:
wip: eks e2e changes

Signed-off-by: Richard Case <richard.case@outlook.com>

vincepri committed:
Log according to sig-instrumentation guidelines

See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#what-method-to-use

Signed-off-by: Vince Prignano <vincepri@redhat.com>

richardcase committed:
chore: fix eks e2e

After the recenet change around the subnets the e2e where breaking as
it wasn't using the new resource id field.

Signed-off-by: Richard Case <richard.case@outlook.com>

k8s-ci-robot committed:
Merge pull request #4571 from vincepri/security-group-tag-owned

When tagging security groups, ensure cloud provider tag is only on lb

vincepri committed:
When using an s3 bucket, add a vpc endpoint

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4563 from vincepri/add-tcp-nlb

:sparkles: Support additional TCP listeners on NLBs

k8s-ci-robot committed:
Merge pull request #4567 from vincepri/gc-earlier

Try to delete as many resources as possible, then requeue

Skarlso committed:
fix: sorting the tags list to make it determined

k8s-ci-robot committed:
Merge pull request #4579 from Skarlso/fix-tag-order

fix: sorting the tags list to make it determined

k8s-ci-robot committed:
Merge pull request #4552 from vincepri/allow-esp-port

:seedling: Allow ESP protocol to be set in IngressRules

k8s-ci-robot committed:
Merge pull request #4536 from richardcase/cameron_reviewer

[PROPOSAL] Reviewer and maintainer changes

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.373 to 0.1.374.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.373...v0.1.374)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4580 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.374

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.373 to 0.1.374

dependabot[bot] committed:
build(deps): bump actions/checkout from 4.1.0 to 4.1.1

Bumps [actions/checkout](https://github.com/actions/checkout) from 4.1.0 to 4.1.1.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4.1.0...v4.1.1)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

AndiDog committed:
Fix launch template reconciliation if bootstrap data secret cannot be read

k8s-ci-robot committed:
Merge pull request #4576 from vincepri/logger-guidelines

⚠️ Log according to sig-instrumentation guidelines

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.374 to 0.1.377.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.374...v0.1.377)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.56.1 to 1.56.3

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.56.1 to 1.56.3.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.56.1...v1.56.3)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4588 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.377

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.374 to 0.1.377

dependabot[bot] committed:
build(deps): bump github.com/go-logr/logr from 1.2.4 to 1.3.0

Bumps [github.com/go-logr/logr](https://github.com/go-logr/logr) from 1.2.4 to 1.3.0.
- [Release notes](https://github.com/go-logr/logr/releases)
- [Changelog](https://github.com/go-logr/logr/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-logr/logr/compare/v1.2.4...v1.3.0)

---
updated-dependencies:
- dependency-name: github.com/go-logr/logr
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

wyike committed:
Update CAPA supported AMI OS

k8s-ci-robot committed:
Merge pull request #4597 from wyike/update-ami-os

Update CAPA supported AMI OS

dependabot[bot] committed:
build(deps): bump github.com/onsi/gomega from 1.28.0 to 1.29.0

Bumps [github.com/onsi/gomega](https://github.com/onsi/gomega) from 1.28.0 to 1.29.0.
- [Release notes](https://github.com/onsi/gomega/releases)
- [Changelog](https://github.com/onsi/gomega/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/gomega/compare/v1.28.0...v1.29.0)

---
updated-dependencies:
- dependency-name: github.com/onsi/gomega
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4581 from kubernetes-sigs/dependabot/github_actions/actions/checkout-4.1.1

build(deps): bump actions/checkout from 4.1.0 to 4.1.1

k8s-ci-robot committed:
Merge pull request #4593 from kubernetes-sigs/dependabot/go_modules/google.golang.org/grpc-1.56.3

build(deps): bump google.golang.org/grpc from 1.56.1 to 1.56.3

k8s-ci-robot committed:
Merge pull request #4594 from kubernetes-sigs/dependabot/go_modules/github.com/onsi/gomega-1.29.0

build(deps): bump github.com/onsi/gomega from 1.28.0 to 1.29.0

k8s-ci-robot committed:
Merge pull request #4596 from kubernetes-sigs/dependabot/go_modules/github.com/go-logr/logr-1.3.0

build(deps): bump github.com/go-logr/logr from 1.2.4 to 1.3.0

richardcase committed:
chore: maintainer & reviewer changes

Signed-off-by: Richard Case <richard.case@outlook.com>

dependabot[bot] committed:
build(deps): bump github.com/docker/docker in /hack/tools

Bumps [github.com/docker/docker](https://github.com/docker/docker) from 23.0.5+incompatible to 24.0.7+incompatible.
- [Release notes](https://github.com/docker/docker/releases)
- [Commits](https://github.com/docker/docker/compare/v23.0.5...v24.0.7)

---
updated-dependencies:
- dependency-name: github.com/docker/docker
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/docker/docker

Bumps [github.com/docker/docker](https://github.com/docker/docker) from 24.0.5+incompatible to 24.0.7+incompatible.
- [Release notes](https://github.com/docker/docker/releases)
- [Commits](https://github.com/docker/docker/compare/v24.0.5...v24.0.7)

---
updated-dependencies:
- dependency-name: github.com/docker/docker
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4575 from richardcase/fix_eks_e2e

fix: eks e2e test failing as subnet name was used

dependabot[bot] committed:
build(deps): bump sigs.k8s.io/yaml from 1.3.0 to 1.4.0

Bumps [sigs.k8s.io/yaml](https://github.com/kubernetes-sigs/yaml) from 1.3.0 to 1.4.0.
- [Release notes](https://github.com/kubernetes-sigs/yaml/releases)
- [Changelog](https://github.com/kubernetes-sigs/yaml/blob/master/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/yaml/compare/v1.3.0...v1.4.0)

---
updated-dependencies:
- dependency-name: sigs.k8s.io/yaml
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

fiunchinho committed:
Filter out reconciling events on externally managed AWSManagedClusters

k8s-ci-robot committed:
Merge pull request #4601 from kubernetes-sigs/dependabot/go_modules/github.com/docker/docker-24.0.7incompatible

build(deps): bump github.com/docker/docker from 24.0.5+incompatible to 24.0.7+incompatible

k8s-ci-robot committed:
Merge pull request #4600 from kubernetes-sigs/dependabot/go_modules/hack/tools/github.com/docker/docker-24.0.7incompatible

build(deps): bump github.com/docker/docker from 23.0.5+incompatible to 24.0.7+incompatible in /hack/tools

k8s-ci-robot committed:
Merge pull request #4603 from kubernetes-sigs/dependabot/go_modules/sigs.k8s.io/yaml-1.4.0

build(deps): bump sigs.k8s.io/yaml from 1.3.0 to 1.4.0

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc in /hack/tools

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.55.0 to 1.56.3.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.55.0...v1.56.3)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4592 from kubernetes-sigs/dependabot/go_modules/hack/tools/google.golang.org/grpc-1.56.3

build(deps): bump google.golang.org/grpc from 1.55.0 to 1.56.3 in /hack/tools

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.377 to 0.1.379.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.377...v0.1.379)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4606 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.379

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.377 to 0.1.379

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.379 to 0.1.380.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.379...v0.1.380)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4598 from richardcase/contributir_changes

[PROPOSAL] move fiunchinho to reviewer

k8s-ci-robot committed:
Merge pull request #4609 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.380

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.379 to 0.1.380

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.380 to 0.1.381.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.380...v0.1.381)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Ankitasw committed:
Bump dependencies manually

k8s-ci-robot committed:
Merge pull request #4611 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.381

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.380 to 0.1.381

k8s-ci-robot committed:
Merge pull request #4589 from AndiDog/machine-pool-bootstrap-config-ref-rotation

Fix launch template reconciliation if bootstrap data secret cannot be read

dependabot[bot] committed:
build(deps): bump k8s.io/klog/v2 from 2.100.1 to 2.110.1 in /hack/tools

Bumps [k8s.io/klog/v2](https://github.com/kubernetes/klog) from 2.100.1 to 2.110.1.
- [Release notes](https://github.com/kubernetes/klog/releases)
- [Changelog](https://github.com/kubernetes/klog/blob/main/RELEASE.md)
- [Commits](https://github.com/kubernetes/klog/compare/v2.100.1...v2.110.1)

---
updated-dependencies:
- dependency-name: k8s.io/klog/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump golang.org/x/text from 0.13.0 to 0.14.0

Bumps [golang.org/x/text](https://github.com/golang/text) from 0.13.0 to 0.14.0.
- [Release notes](https://github.com/golang/text/releases)
- [Commits](https://github.com/golang/text/compare/v0.13.0...v0.14.0)

---
updated-dependencies:
- dependency-name: golang.org/x/text
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/spf13/cobra from 1.7.0 to 1.8.0

Bumps [github.com/spf13/cobra](https://github.com/spf13/cobra) from 1.7.0 to 1.8.0.
- [Release notes](https://github.com/spf13/cobra/releases)
- [Commits](https://github.com/spf13/cobra/compare/v1.7.0...v1.8.0)

---
updated-dependencies:
- dependency-name: github.com/spf13/cobra
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndiDog committed:
Pass logger to machine pool scope to get verbosity right, fix copying logger in `With...` methods

k8s-ci-robot committed:
Merge pull request #4577 from vincepri/s3-endpoint

When using an s3 bucket, add a vpc endpoint

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.381 to 0.1.383.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.381...v0.1.383)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4621 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.383

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.381 to 0.1.383

k8s-ci-robot committed:
Merge pull request #4614 from kubernetes-sigs/dependabot/go_modules/golang.org/x/text-0.14.0

build(deps): bump golang.org/x/text from 0.13.0 to 0.14.0

k8s-ci-robot committed:
Merge pull request #4617 from kubernetes-sigs/dependabot/go_modules/github.com/spf13/cobra-1.8.0

build(deps): bump github.com/spf13/cobra from 1.7.0 to 1.8.0

dependabot[bot] committed:
build(deps): bump golang.org/x/crypto from 0.14.0 to 0.15.0

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.14.0 to 0.15.0.
- [Commits](https://github.com/golang/crypto/compare/v0.14.0...v0.15.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.383 to 0.1.385.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.383...v0.1.385)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4613 from kubernetes-sigs/dependabot/go_modules/hack/tools/k8s.io/klog/v2-2.110.1

build(deps): bump k8s.io/klog/v2 from 2.100.1 to 2.110.1 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4623 from kubernetes-sigs/dependabot/go_modules/golang.org/x/crypto-0.15.0

build(deps): bump golang.org/x/crypto from 0.14.0 to 0.15.0

k8s-ci-robot committed:
Merge pull request #4624 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.385

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.383 to 0.1.385

dependabot[bot] committed:
build(deps): bump github.com/onsi/gomega from 1.29.0 to 1.30.0

Bumps [github.com/onsi/gomega](https://github.com/onsi/gomega) from 1.29.0 to 1.30.0.
- [Release notes](https://github.com/onsi/gomega/releases)
- [Changelog](https://github.com/onsi/gomega/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/gomega/compare/v1.29.0...v1.30.0)

---
updated-dependencies:
- dependency-name: github.com/onsi/gomega
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

AndiDog committed:
Use `AdditionalTags` for S3 buckets

k8s-ci-robot committed:
Merge pull request #4625 from AndiDog/s3-bucket-additional-tags

Use `AdditionalTags` for S3 buckets

k8s-ci-robot committed:
Merge pull request #4604 from giantswarm/eks-filter-managed-infra

Filter out reconciling events on externally managed AWSManagedClusters

Ankitasw committed:
Bump CAPI to v1.5.3

k8s-ci-robot committed:
Merge pull request #4612 from Ankitasw/bump-deps

Bump golangci-lint dependency

k8s-ci-robot committed:
Merge pull request #4622 from kubernetes-sigs/dependabot/go_modules/github.com/onsi/gomega-1.30.0

build(deps): bump github.com/onsi/gomega from 1.29.0 to 1.30.0

dependabot[bot] committed:
build(deps): bump github.com/mikefarah/yq/v4 in /hack/tools

Bumps [github.com/mikefarah/yq/v4](https://github.com/mikefarah/yq) from 4.35.2 to 4.40.1.
- [Release notes](https://github.com/mikefarah/yq/releases)
- [Changelog](https://github.com/mikefarah/yq/blob/master/release_notes.txt)
- [Commits](https://github.com/mikefarah/yq/compare/v4.35.2...v4.40.1)

---
updated-dependencies:
- dependency-name: github.com/mikefarah/yq/v4
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/onsi/ginkgo/v2 from 2.13.0 to 2.13.1

Bumps [github.com/onsi/ginkgo/v2](https://github.com/onsi/ginkgo) from 2.13.0 to 2.13.1.
- [Release notes](https://github.com/onsi/ginkgo/releases)
- [Changelog](https://github.com/onsi/ginkgo/blob/master/CHANGELOG.md)
- [Commits](https://github.com/onsi/ginkgo/compare/v2.13.0...v2.13.1)

---
updated-dependencies:
- dependency-name: github.com/onsi/ginkgo/v2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

kranurag7 committed:
add verifier action for pull requests

This commit adds a verifier action that will check the title of the
pull request.

Signed-off-by: Anurag <81210977+kranurag7@users.noreply.github.com>

k8s-ci-robot committed:
Merge pull request #4631 from kubernetes-sigs/dependabot/go_modules/hack/tools/github.com/mikefarah/yq/v4-4.40.1

build(deps): bump github.com/mikefarah/yq/v4 from 4.35.2 to 4.40.1 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4632 from kubernetes-sigs/dependabot/go_modules/github.com/onsi/ginkgo/v2-2.13.1

build(deps): bump github.com/onsi/ginkgo/v2 from 2.13.0 to 2.13.1

dependabot[bot] committed:
build(deps): bump actions/github-script from 6.4.1 to 7.0.0

Bumps [actions/github-script](https://github.com/actions/github-script) from 6.4.1 to 7.0.0.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/d7906e4ad0b1822421a7e6a35d5ca353c962f410...e69ef5462fd455e02edcaf4dd7708eda96b9eda0)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4635 from kubernetes-sigs/dependabot/github_actions/actions/github-script-7.0.0

build(deps): bump actions/github-script from 6.4.1 to 7.0.0

vincepri committed:
:bug: e2e should delete the cluster before validating

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4636 from vincepri/fix-e2e-vp

:bug: e2e should delete the cluster before validating

dependabot[bot] committed:
build(deps): bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.385 to 0.1.386.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.385...v0.1.386)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4638 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.386

build(deps): bump github.com/openshift-online/ocm-sdk-go from 0.1.385 to 0.1.386

vincepri committed:
:bug: Fix race condition between cluster delete, and check

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4639 from vincepri/fix-e2e-vp2

:bug: Fix race condition between cluster delete, and check

dependabot[bot] committed:
build(deps): bump actions/github-script from 7.0.0 to 7.0.1

Bumps [actions/github-script](https://github.com/actions/github-script) from 7.0.0 to 7.0.1.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/e69ef5462fd455e02edcaf4dd7708eda96b9eda0...60a0d83039c74a4aee543508d2ffcb1c3799cdea)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/mikefarah/yq/v4 in /hack/tools

Bumps [github.com/mikefarah/yq/v4](https://github.com/mikefarah/yq) from 4.40.1 to 4.40.2.
- [Release notes](https://github.com/mikefarah/yq/releases)
- [Changelog](https://github.com/mikefarah/yq/blob/master/release_notes.txt)
- [Commits](https://github.com/mikefarah/yq/compare/v4.40.1...v4.40.2)

---
updated-dependencies:
- dependency-name: github.com/mikefarah/yq/v4
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4640 from kubernetes-sigs/dependabot/github_actions/actions/github-script-7.0.1

build(deps): bump actions/github-script from 7.0.0 to 7.0.1

k8s-ci-robot committed:
Merge pull request #4641 from kubernetes-sigs/dependabot/go_modules/hack/tools/github.com/mikefarah/yq/v4-4.40.2

build(deps): bump github.com/mikefarah/yq/v4 from 4.40.1 to 4.40.2 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4627 from Ankitasw/bump-capi

Bump CAPI to v1.5.3

dependabot[bot] committed:
build(deps): bump google.golang.org/grpc from 1.58.2 to 1.58.3

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.58.2 to 1.58.3.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.58.2...v1.58.3)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump github.com/aws/amazon-vpc-cni-k8s

Bumps [github.com/aws/amazon-vpc-cni-k8s](https://github.com/aws/amazon-vpc-cni-k8s) from 1.15.0 to 1.15.4.
- [Release notes](https://github.com/aws/amazon-vpc-cni-k8s/releases)
- [Changelog](https://github.com/aws/amazon-vpc-cni-k8s/blob/master/CHANGELOG.md)
- [Commits](https://github.com/aws/amazon-vpc-cni-k8s/compare/v1.15.0...v1.15.4)

---
updated-dependencies:
- dependency-name: github.com/aws/amazon-vpc-cni-k8s
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4644 from kubernetes-sigs/dependabot/go_modules/google.golang.org/grpc-1.58.3

build(deps): bump google.golang.org/grpc from 1.58.2 to 1.58.3

k8s-ci-robot committed:
Merge pull request #4646 from kubernetes-sigs/dependabot/go_modules/github.com/aws/amazon-vpc-cni-k8s-1.15.4

build(deps): bump github.com/aws/amazon-vpc-cni-k8s from 1.15.0 to 1.15.4

dependabot[bot] committed:
build(deps): bump github.com/go-jose/go-jose/v3 in /hack/tools

Bumps [github.com/go-jose/go-jose/v3](https://github.com/go-jose/go-jose) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/go-jose/go-jose/releases)
- [Changelog](https://github.com/go-jose/go-jose/blob/v3/CHANGELOG.md)
- [Commits](https://github.com/go-jose/go-jose/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: github.com/go-jose/go-jose/v3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4649 from kubernetes-sigs/dependabot/go_modules/hack/tools/github.com/go-jose/go-jose/v3-3.0.1

build(deps): bump github.com/go-jose/go-jose/v3 from 3.0.0 to 3.0.1 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4634 from kranurag7/kr/add-pr-verify

🌱 Add verifier action for pull requests

richardcase committed:
chore: update metadata for 2.3.0 release

Signed-off-by: Richard Case <richard.case@outlook.com>

kranurag7 committed:
use seedling emoji for dependabot PRs

this commit adds seedling prefix in the PR title whenever dependabot
opens a PR to update deps.

Signed-off-by: Anurag <81210977+kranurag7@users.noreply.github.com>

k8s-ci-robot committed:
Merge pull request #4651 from richardcase/metadata_23

✨chore: update metadata for 2.3.0 release

k8s-ci-robot committed:
Merge pull request #4652 from kranurag7/kr/use-seedling-prefix-for-bot

🌱 use seedling emoji for dependabot PRs

calvix committed:
Update ASG - do not set desired value for machinepool which have externally managed replicas

fiunchinho committed:
Prevent users setting values that are invalid configuration on AWS

fiunchinho committed:
Update exp/api/v1beta2/awsmachinepool_webhook.go

Co-authored-by: Andreas Sommer <andreas@giantswarm.io>

fiunchinho committed:
Don't error if ASG is missing when deciding to refresh instances

k8s-ci-robot committed:
Merge pull request #4660 from giantswarm/no-error-missing-asg-instance-refresh

:bug: Don't error if ASG is missing when deciding to refresh instances

k8s-ci-robot committed:
Merge pull request #4620 from AndiDog/fix-logging

Pass logger to machine pool scope to get verbosity right, fix copying logger in `With...` methods

k8s-ci-robot committed:
Merge pull request #4656 from giantswarm/validate-spot-mixed-instances

:sparkles: Prevent users setting values that are invalid configuration on AWS

fiunchinho committed:
There is no need to check instance refresh if ASG is not found

Co-authored-by: Andreas Sommer <andreas@giantswarm.io>

k8s-ci-robot committed:
Merge pull request #4654 from calvix/do-not-set-desired-when-updating-asg-with-external-scaler

:bug: Update ASG - do not set desired value for machinepool which have externally managed replicas

muraee committed:
Add ROSA feature-gate to manager.yaml

muraee committed:
Add ROSA CRDs to kustomization.yaml

muraee committed:
Fixed rosacontrolplane_controller permissions

thefirstofthe300 committed:
cleanup: eliminate log spam when using S3 secrets

mjlshen committed:
Update awsmachine providerID and instanceID immediately after ec2:RunInstances is called

This mitigates issues caused by falling back to tag-based searching for
EC2 instances in case future AWS calls fail, such as attaching ENIs to
security groups or tagging ENIs.

Signed-off-by: Michael Shen <mishen@umich.edu>

dependabot[bot] committed:
:seedling: bump github.com/openshift-online/ocm-sdk-go

Bumps [github.com/openshift-online/ocm-sdk-go](https://github.com/openshift-online/ocm-sdk-go) from 0.1.386 to 0.1.388.
- [Release notes](https://github.com/openshift-online/ocm-sdk-go/releases)
- [Changelog](https://github.com/openshift-online/ocm-sdk-go/blob/main/CHANGES.md)
- [Commits](https://github.com/openshift-online/ocm-sdk-go/compare/v0.1.386...v0.1.388)

---
updated-dependencies:
- dependency-name: github.com/openshift-online/ocm-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

fiunchinho committed:
Enable transit encryption to S3 bucket

nrb committed:
Use external CCM in default template

Copies the structure of the
cluster-template-external-cloud-provider.yaml into the default template,
with the folloiwng changes.

* The following version updates were made:
    aws-cloud-controller-manager: v1.28.3
    CSI driver:                   v1.25.0
    CSI provisioner:              v3.6.2
    CSI attacher:                 v4.4.2
    CSI resizer:                  v1.9.2
    CSI livenessprobe:            v2.11.0
    CSI register:                 v2.9.2

* The CNI references were removed, since the Quickstart lets the user
install their own CNI.

* Updated the contents of the aws-ccm-external.yaml file based on files in
https://github.com/kubernetes/cloud-provider-aws/tree/master/examples/existing-cluster/base.
** Changed `--configure-cloud-routes=false` from the default
** Changed node selector from `node-role.kubernetes.io/master` to
`node-role.kubernetes.io/controle-plane`.

NOTE: Because this change requires the use of ClusterResourceSet, the
Quickstart will need to be updated to include `export
EXP_CLUSTER_RESOURCE_SET=true` prior to initializing the management
cluster.

Signed-off-by: Nolan Brubaker <nolan@nbrubaker.com>

AndiDog committed:
:seedling: bump sigs.k8s.io/aws-iam-authenticator from 0.6.12 to 0.6.13

dependabot[bot] committed:
:seedling: bump actions/setup-go from 4 to 5

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 4 to 5.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4674 from kubernetes-sigs/dependabot/go_modules/github.com/openshift-online/ocm-sdk-go-0.1.388

:seedling: bump github.com/openshift-online/ocm-sdk-go from 0.1.386 to 0.1.388

k8s-ci-robot committed:
Merge pull request #4670 from mjlshen/4629-status

:bug: Update awsmachine providerID and instanceID immediately after ec2:RunInstances is called

harveyxia committed:
Wrap error when failing to get infra provider cluster

enxebre committed:
Add ROSAControlPlaneReadyCondition, ocmclient and WorkerRoleARN field

    This commit introduces several improvements to the rosa control plane:
    - Add ROSAControlPlaneReadyCondition
    - Add helpers for ocmclient
    - Add WorkerRoleARN field to the API to satisfy latest ocm API requirements

richardcase committed:
chore: add andreas as a reviewer

Signed-off-by: Richard Case <richard.case@outlook.com>

richardcase committed:
chore: update dependabot to group updates

Signed-off-by: Richard Case <richard.case@outlook.com>

k8s-ci-robot committed:
Merge pull request #4662 from giantswarm/ignore-notfound-asg

:bug:  There is no need to check instance refresh if ASG is not found

k8s-ci-robot committed:
Merge pull request #4682 from kubernetes-sigs/dependabot/github_actions/actions/setup-go-5

:seedling: bump actions/setup-go from 4 to 5

k8s-ci-robot committed:
Merge pull request #4687 from richardcase/adreas_reviewer

🐛 [PROPOSAL] add andreas as a reviewer

k8s-ci-robot committed:
Merge pull request #4676 from giantswarm/s3-transit-encryption

:sparkles: Enable transit encryption to S3 bucket

k8s-ci-robot committed:
Merge pull request #4679 from AndiDog/aws-iam-authenticator-0.6.13

🌱 bump sigs.k8s.io/aws-iam-authenticator from 0.6.12 to 0.6.13

k8s-ci-robot committed:
Merge pull request #4688 from richardcase/dependabot_group

🐛 chore: update dependabot to group updates

dependabot[bot] committed:
:seedling: Bump the dependencies group with 2 updates

Bumps the dependencies group with 2 updates: [github/codeql-action](https://github.com/github/codeql-action) and [kubernetes-sigs/kubebuilder-release-tools](https://github.com/kubernetes-sigs/kubebuilder-release-tools).


Updates `github/codeql-action` from 2 to 3
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v2...v3)

Updates `kubernetes-sigs/kubebuilder-release-tools` from 0.4.2 to 0.4.3
- [Release notes](https://github.com/kubernetes-sigs/kubebuilder-release-tools/releases)
- [Changelog](https://github.com/kubernetes-sigs/kubebuilder-release-tools/blob/master/RELEASE.md)
- [Commits](https://github.com/kubernetes-sigs/kubebuilder-release-tools/compare/3c3411345eedc489d1022288aa844691e92a9c29...012269a88fa4c034a0acf1ba84c26b195c0dbab4)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: dependencies
- dependency-name: kubernetes-sigs/kubebuilder-release-tools
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

richardcase committed:
chore: change the ignore dependabot updates

Signed-off-by: Richard Case <richard.case@outlook.com>

k8s-ci-robot committed:
Merge pull request #4689 from kubernetes-sigs/dependabot/github_actions/dependencies-e486f348bc

:seedling: Bump the dependencies group with 2 updates

k8s-ci-robot committed:
Merge pull request #4692 from richardcase/dependabot_group

🌱 chore: change the ignore dependabot updates

dependabot[bot] committed:
:seedling: Bump the dependencies group in /hack/tools with 2 updates

Bumps the dependencies group in /hack/tools with 2 updates: [github.com/itchyny/gojq](https://github.com/itchyny/gojq) and [github.com/mikefarah/yq/v4](https://github.com/mikefarah/yq).


Updates `github.com/itchyny/gojq` from 0.12.13 to 0.12.14
- [Release notes](https://github.com/itchyny/gojq/releases)
- [Changelog](https://github.com/itchyny/gojq/blob/main/CHANGELOG.md)
- [Commits](https://github.com/itchyny/gojq/compare/v0.12.13...v0.12.14)

Updates `github.com/mikefarah/yq/v4` from 4.40.2 to 4.40.5
- [Release notes](https://github.com/mikefarah/yq/releases)
- [Changelog](https://github.com/mikefarah/yq/blob/master/release_notes.txt)
- [Commits](https://github.com/mikefarah/yq/compare/v4.40.2...v4.40.5)

---
updated-dependencies:
- dependency-name: github.com/itchyny/gojq
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: dependencies
- dependency-name: github.com/mikefarah/yq/v4
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4693 from kubernetes-sigs/dependabot/go_modules/hack/tools/dependencies-b61036ebd8

:seedling: Bump the dependencies group in /hack/tools with 2 updates

k8s-ci-robot committed:
Merge pull request #4678 from nrb/fix-4677

🐛: Use external CCM in default template

tobiasgiese committed:
Fix deregistering of deleted CAPI Machines

We have an issue that there is a gap between the CAPI Machine deletion and the AWSMachine deletion.
In this timeframe the kube-apiserver to the to be deleted control plane is no longer possible, but the deregistering of the LB member will be performed only for AWSMachine deletions.
This PR fixes the gap and also deregisteres the instance if the CAPI Machine is deleted.

Signed-off-by: Tobias Giese <tobias.giese@mercedes-benz.com>

dependabot[bot] committed:
:seedling: Bump golang.org/x/crypto from 0.15.0 to 0.17.0

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.15.0 to 0.17.0.
- [Commits](https://github.com/golang/crypto/compare/v0.15.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4701 from mercedes-benz/tobiasgiese/fix-deregister-scaledown

🐛 Fix deregistering of deleted CAPI Machines

fiunchinho committed:
Remove ingress and egress rules from vpc default security group

AndiDog committed:
Log full ARN in GC error messages

fiunchinho committed:
Use go 1.21.5, fix kubectl version detection after `--short` parameter was removed

Co-authored-by: Andreas Sommer <andreas@giantswarm.io>

muraee committed:
Add docs for ROSA

- add new cluster template for ROSA clusters

k8s-ci-robot committed:
Merge pull request #4710 from giantswarm/upgrade-go-1-21

✨ Use go 1.21.5, fix kubectl version detection after `--short` parameter was removed

dependabot[bot] committed:
:seedling: Bump github.com/go-git/go-git/v5 in /hack/tools

Bumps [github.com/go-git/go-git/v5](https://github.com/go-git/go-git) from 5.8.1 to 5.11.0.
- [Release notes](https://github.com/go-git/go-git/releases)
- [Commits](https://github.com/go-git/go-git/compare/v5.8.1...v5.11.0)

---
updated-dependencies:
- dependency-name: github.com/go-git/go-git/v5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

AndiDog committed:
Fix some logging and error cases

k8s-ci-robot committed:
Merge pull request #4665 from muraee/rosa-capi-follow-up

🌱ROSA Cleanup & Documentation

muraee committed:
Add RosaClusterName field to ROSAControlPlane

- ensure RosaClusterName is valid using kubebuild validation
- moved ocmClient to a seperate package and renamed to rosaClient
- updated cluster-template-rosa.yaml
- set ControlPlane.Status.Initialized
- requeue ROSAControlPlane to poll cluster status until ready

muraee committed:
Add ROSAMachinePool CRD & Controller

This introduces basic support to create/delete ROSAMachinePools
Lifecycle is captured in RosaMchinePoolReady condition

- add cluster-template-rosa-machinepool.yaml

k8s-ci-robot committed:
Merge pull request #4709 from giantswarm/gc-full-arn

✨ Log full ARN in GC error messages

dependabot[bot] committed:
:seedling: Bump github.com/cloudflare/circl in /hack/tools

Bumps [github.com/cloudflare/circl](https://github.com/cloudflare/circl) from 1.3.3 to 1.3.7.
- [Release notes](https://github.com/cloudflare/circl/releases)
- [Commits](https://github.com/cloudflare/circl/compare/v1.3.3...v1.3.7)

---
updated-dependencies:
- dependency-name: github.com/cloudflare/circl
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

k8s-ci-robot committed:
Merge pull request #4716 from kubernetes-sigs/dependabot/go_modules/hack/tools/github.com/go-git/go-git/v5-5.11.0

:seedling: Bump github.com/go-git/go-git/v5 from 5.8.1 to 5.11.0 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4685 from harveyxia/awsmachinepool-wrap-error

:bug: Wrap error when AWSMachinePool reconciler fails to get infra provider cluster

k8s-ci-robot committed:
Merge pull request #4702 from kubernetes-sigs/dependabot/go_modules/golang.org/x/crypto-0.17.0

:seedling: Bump golang.org/x/crypto from 0.15.0 to 0.17.0

k8s-ci-robot committed:
Merge pull request #4727 from kubernetes-sigs/dependabot/go_modules/hack/tools/github.com/cloudflare/circl-1.3.7

:seedling: Bump github.com/cloudflare/circl from 1.3.3 to 1.3.7 in /hack/tools

k8s-ci-robot committed:
Merge pull request #4500 from Fedosin/gc_config_cli

Allow to configure garbage collector using clusterawsadm

AndiDog committed:
Skip AWSCluster deletion reconciliation once CAPA finalizer is gone

k8s-ci-robot committed:
Merge pull request #4667 from thefirstofthe300/reduce-log-spam

:bug: cleanup: eliminate log spam when using S3 secrets

k8s-ci-robot committed:
Merge pull request #4724 from AndiDog/logging-fixes

🌱 Fix some logging and error cases

vincepri committed:
S3 bucket region creation, ensure transport policy w/ presigned URLs

Currently the S3 bucket is always created in us-east-1, regardless of
where the cluster is located. In addition, this PR always ensures that
we add a policy to ensure transport encryption is enabled, even when
using presigned URLs.

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4707 from giantswarm/secure-default-vpc-sg

:sparkles: Remove ingress and egress rules from vpc default security group

muraee committed:
read credentials(token) from a referenced secret

muraee committed:
fixed unit tests

- reset CommandLine flagSet before calling klog.InitFlags(nil) to avoid conflicts if an imported package already called it.

k8s-ci-robot committed:
Merge pull request #4725 from muraee/rosa-machine-mgmt

✨ ROSA machinePools support

k8s-ci-robot committed:
Merge pull request #4730 from giantswarm/delete-reconcile-only-with-finalizer

✨ Skip AWSCluster deletion reconciliation once CAPA finalizer is gone

k8s-ci-robot committed:
Merge pull request #4732 from vincepri/bucket-region-andpolicy

:bug: S3 Bucket should be created in the same region, always add transport encryption policy

damdo committed:
✨ Bump CAPI to v1.6.x (#4739)

* use k8s.io/utils/ptr instead of pointer

* reference local copy of noderefutil

following the deprecation of noderefutil, the package has now been removed from cluster-api

* bump deps for capi v1.6.1, fix changes

* e2e: fix pointer panic in RefreshVPCState

* fix: remove unnecessary fmt

* wip: set patch version Go 1.21.5 for netlify

* wip: add debugging on VPC creation failure

AverageMarcus committed:
Move AverageMarcus to emeritus status

Signed-off-by: Marcus Noble <github@marcusnoble.co.uk>

dependabot[bot] committed:
:seedling: Bump the dependencies group with 1 update

Bumps the dependencies group with 1 update: [actions/cache](https://github.com/actions/cache).


Updates `actions/cache` from 3 to 4
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
:seedling: Bump the dependencies group in /hack/tools with 1 update

Bumps the dependencies group in /hack/tools with 1 update: [github.com/joelanford/go-apidiff](https://github.com/joelanford/go-apidiff).


Updates `github.com/joelanford/go-apidiff` from 0.7.0 to 0.8.2
- [Release notes](https://github.com/joelanford/go-apidiff/releases)
- [Commits](https://github.com/joelanford/go-apidiff/compare/v0.7.0...v0.8.2)

---
updated-dependencies:
- dependency-name: github.com/joelanford/go-apidiff
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

vincepri committed:
:book: Add /agenda redirect to book

Adds a fixed-url redirect https://cluster-api-aws.sigs.k8s.io/agenda
to use as a meeting invite.

Signed-off-by: Vince Prignano <vincepri@redhat.com>

k8s-ci-robot committed:
Merge pull request #4751 from giantswarm/AverageMarcus_emeritus

🌱 Move AverageMarcus to emeritus status

k8s-ci-robot committed:
Merge pull request #4752 from kubernetes-sigs/dependabot/github_actions/dependencies-3e83326924

:seedling: Bump the dependencies group with 1 update

k8s-ci-robot committed:
Merge pull request #4753 from kubernetes-sigs/dependabot/go_modules/hack/tools/dependencies-82a3292ea4

:seedling: Bump the dependencies group in /hack/tools with 1 update

k8s-ci-robot committed:
Merge pull request #4756 from vincepri/agenda-redirect

:book: Add /agenda redirect to book
